Patch-Source: https://gitlab.archlinux.org/archlinux/packaging/packages/protobuf/-/blob/e93df92fead2d46cc2825a866af3f402d498439c/soversion.patch
use an soversion that actually makes sense and not just the full pkgver
--
diff --git a/cmake/libprotobuf-lite.cmake b/cmake/libprotobuf-lite.cmake
index 9aa81fb88..f599940cc 100644
--- a/cmake/libprotobuf-lite.cmake
+++ b/cmake/libprotobuf-lite.cmake
@@ -37,6 +37,7 @@ if(protobuf_BUILD_SHARED_LIBS)
 endif()
 set_target_properties(libprotobuf-lite PROPERTIES
     VERSION ${protobuf_VERSION}
+    SOVERSION ${protobuf_VERSION_MINOR}
     OUTPUT_NAME ${LIB_PREFIX}protobuf-lite
     DEBUG_POSTFIX "${protobuf_DEBUG_POSTFIX}"
     # For -fvisibility=hidden and -fvisibility-inlines-hidden
diff --git a/cmake/libprotobuf.cmake b/cmake/libprotobuf.cmake
index 11c09b1bc..0d324f28e 100644
--- a/cmake/libprotobuf.cmake
+++ b/cmake/libprotobuf.cmake
@@ -39,6 +39,7 @@ if(protobuf_BUILD_SHARED_LIBS)
 endif()
 set_target_properties(libprotobuf PROPERTIES
     VERSION ${protobuf_VERSION}
+    SOVERSION ${protobuf_VERSION_MINOR}
     OUTPUT_NAME ${LIB_PREFIX}protobuf
     DEBUG_POSTFIX "${protobuf_DEBUG_POSTFIX}"
     # For -fvisibility=hidden and -fvisibility-inlines-hidden
diff --git a/cmake/libprotoc.cmake b/cmake/libprotoc.cmake
index 1fd275d5a..13d061c57 100644
--- a/cmake/libprotoc.cmake
+++ b/cmake/libprotoc.cmake
@@ -27,6 +27,7 @@ endif()
 set_target_properties(libprotoc PROPERTIES
     COMPILE_DEFINITIONS LIBPROTOC_EXPORTS
     VERSION ${protobuf_VERSION}
+    SOVERSION ${protobuf_VERSION_MINOR}
     OUTPUT_NAME ${LIB_PREFIX}protoc
     DEBUG_POSTFIX "${protobuf_DEBUG_POSTFIX}"
     # For -fvisibility=hidden and -fvisibility-inlines-hidden
